### PR TITLE
Add positive quirk - Linguist 

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
This adds a new positive perk called Linguist that lets a player pick one language (or random) from the character preference menu.  The following languages can be learned:

- **Uncommon**
- **Nekomimetic**
- **Draconic**
- **Moffic**
- **Calcic** _(need Plasmaman tongue to speak)_
- **Voltaic** _(need Ethereal tongue to speak)_

If a player already knows the selected language, then a random one is selected instead.  If you are the curator, then nothing happens since they know every language already.

## Why It's Good For The Game
Encourages more language use and variety.  For a positive quirk, this falls into the non-combat category and I think is useful enough for people to want to take it.

## Changelog
:cl:
add: New positive quirk - Linguist.  You can select one of the following languages to learn: Uncommon, Nekomimetic, Draconic, Moffic, Calcic, Voltaic. However both Calcic and Voltaic requires the appropriate race tongue to be able to speak, but you can still understand and hear others speaking.
/:cl:

### CODE REVIEW

File: code/datums/quirks/good.dm [hunk 1]
```diff
@@ -293,3 +293,58 @@
...
+		if("Nekomimetic")
+			language = /datum/language/nekomimetic
+		if("Draconic")
+			language = /datum/language/draconic
+		if("Moffic")
+			language = /datum/language/moffic
+		if("Calcic") // only plasmamen can speak
+			language = /datum/language/calcic
+		if("Uncommon")
+			language = /datum/language/uncommon
```
Comment (Fikou) [inline code/datums/quirks/good.dm:332]: turn roundstart languages into an associated list

**Comment by thgvr** [comment]
Can't you just buy a language translation key from Cargo?

**Comment by Mothblocks** [comment]
players hate going to cargo for anything so yes but also no

**Comment by Atlatl1807** [comment]
> Can't you just buy a language translation key from Cargo?

Yes but this still gives other advantages. The quirk gives you the convenience of not having to buy the key. It also prevents your ability to speak the language from being emped or stolen. 

You could just get an instrument from the stage instead of taking the musician quirk. But people take the musician quirk because it gives you a mood bonus and a convenient instrument teleporter.

**Comment by optimumtact** [comment]
The only problem here is that it continues the slow slide of cargo into irrelevance.

**Comment by SkeletalElite** [comment]
Okay if this is making cargo irrelevant, how many times in the past month has the translator been purchased

**Review by MrMelbert — COMMENTED** [review]
it's worth noting that, because roundstart races are config based, calling it "`roundstart_languages`" is misleading

**Comment by timothymtorres** [comment]
List of ways in game for people to translate languages:
- Curator/AI/Cyborgs
- Book of babel (lavaland loot)
- Language translation keys from cargo
- Voice recorders (available roundstart and at autolathe)
- Translator circuits (available roundstart and is prebuilt)
- pAIs

Having it available as a quirk that costs 6 points that lets you translate _one_ language is pretty reasonable.  Especially considering that most of the above options, translates **ALL** languages at roundstart for no cost.  I remember trying to make #68079 so that voice recorders would have a more severe limitation since IMO it was a cheap exploit to bypass language & tongue restriction mechanics.

**Comment by thgvr** [comment]
I think it's pretty lame as some of the only benefits for some races is having the language that others don't have roundstart access too, and quirking that out of the game feels awful to me.

**Comment by newfren** [comment]
6 quirk points is a lot of quirk points, I wouldn't expect to see this very often.

**Comment by Serxule** [comment]
yeah 6 points is a lot for something thats so insignificant

File: code/datums/quirks/good.dm (hunk 1 above)
Comment (timothymtorres) [reply code/datums/quirks/good.dm]: Done

**Comment by timothymtorres** [comment]
I lowered the cost to 4.

**Comment by optimumtact** [comment]
now it's even less mergeable

**Comment by Overuseddart** [comment]
Negative quirk that makes you speak a random language when? (Out of all of them so its better, you WILL speak golem)

**Comment by Mickyan** [comment]
I think as a compromise to not encroach too much upon other ways to do this during the round you could make it so it grants the ability to understand the language but not speak it (set spoken = FALSE in grant_language(), afaik), or even split that into two quirks with different point cost

**Comment by ShizCalev** [comment]
quite cunning.

**Comment by JohnFulpWillard** [comment]
I disagree with this PR, asking a Curator to translate, or getting a pAI, are scenarios where you have to go out of your way to involve other people. Getting an encryption key from Cargo works the same way and I would prefer people use that instead, being a quirk makes experiences with languages less unique (such as people trying to hide conversation by speaking a new language, intercepted by a Curator or Book of Babel or encryption key).

**Pull request closed without merging**